### DISCUSSION TITLE
Whether the Parent stores information about the Child persistently?

### DISCUSSION THREAD

**Discussion by jinpeng1989** [openthread/openthread#8869] [Q&A] [answered]
Whether the Parent stores information about the Child persistently? Such as the Mode, Address Registration, Timeout of the Child?

**Comment by abtink** [comment, marked as answer]
Parent stores a lot of information about child in its child table. Check `common/topology.h` and `Neighbor` and `Child` class definitions.

If your question  about info saved in non-volatile settings (available upon parent reset/reboot to restore its previous children) then check the `Settings.hpp` header and `Settings::ChildInfo` definition:
```
        Mac::ExtAddress mExtAddress; ///< Extended Address
        uint32_t        mTimeout;    ///< Timeout
        uint16_t        mRloc16;     ///< RLOC16
        uint8_t         mMode;       ///< The MLE device mode
        uint16_t        mVersion;    ///< Version
```

**Reply by jinpeng1989** [reply]
Whether there is already a relevant discussion on this topic. Please help to provide the links.
Since the Parent stores the information of the Child. Are there some requirements for Child updates the information? I am concerned with the minimum time interval allowed between updates(In particular, modify the timeout time). Are there any requirements from specification or the Openthread Stack? In some chip implementations, there are some problems with writing flash quickly.

**Reply by abtink** [reply]
Thread specification section 10.12 (titled "Persistent Data") describes this.
